### PR TITLE
Disable torch.complex32 dtype

### PR DESCRIPTION
**Pull request by anjali411**
Related https://github.com/pytorch/pytorch/issues/69209#event-5722454832

Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #70606

cc @ezyang @gchanan @anjali411 @dylanbespalko @mruberry @Lezcano @nikitaved

Differential Revision: [D33954861](https://our.internmc.facebook.com/intern/diff/D33954861)

### CODE REVIEW

**Comment by anjali411** [comment]
@ngimel should we just cherry-pick this for the next release?

File: torch/testing/_comparison.py [hunk 1]
```diff
@@ -1106,8 +1105,6 @@ def assert_close(
     +---------------------------+------------+----------+
```
Comment (mruberry) [inline torch/testing/_comparison.py:1105]: cc @pmeier

**Review by mruberry — APPROVED** [review]
Cool! (So long as the tests pass.)

Note from offline meeting: @ngimel plans to file a follow-up issue for proper ComplexHalf support

**Comment by anjali411** [comment]
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33954861).

**Comment by anjali411** [comment]
@anjali411 has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33954861).

**Comment by anjali411** [comment]
@JackCaoG I will be cherry-picking this PR in the 1.11 release branch and looks like it'll require an xla side change (remove `torch.complex32` from this list https://github.com/pytorch/xla/blob/3bb039c8e8cff30e23394e47bf67443b0c200100/test/pytorch_test_base.py#L467-L469)

**Comment by JackCaoG** [comment]
@anjali411 Seems like we only need to remove this dtype from our repo, I can do that and backport to our 1.11 branch.

**Comment by malfet** [comment]
@anjali411 correct me if I'm wrong, but https://github.com/pytorch/pytorch/commit/13b403b7144fb99235e8e9c17316274a5a04dc17 has been landed in 1.11, so we probably can remove the milestone (and close this PR) after xla change is landed

**Comment by JackCaoG** [comment]
@malfet I just merge the pt/xla 1.11 pr. Pytroch/xla 1.11 branch is missing several important prs. I am thinking about recut the 1.11 from master which is a bit easier than me manually porting a bunch of pr at the same time. Will let you know if I do that.

**Comment by anjali411** [comment]
> @anjali411 correct me if I'm wrong, but [13b403b](https://github.com/pytorch/pytorch/commit/13b403b7144fb99235e8e9c17316274a5a04dc17) has been landed in 1.11, so we probably can remove the milestone (and close this PR) after xla change is landed

yup absolutely!

**Pull request closed without merging**